Resources/public/js/views/services/ez-trashviewservice.js

```diff
@@ -223,6 +223,60 @@ YUI.add('ez-trashviewservice', function (Y) {
         },
 
         /**
+         * Restores trash items
+         *
+         * @method _restoreItems
+         * @protected
+         * @param {Object} e restoreItems event facade
+         * @param {Array} e.trashItems  List of trashItems to be restored
+         */
+        _restoreItems: function (e) {
+            var loadOptions = {api: this.get('capi')},
+                tasks = new Y.Parallel(),
+                service = this,
+                notificationName = "restoreTrashItems";
+
+            this._notify(
+                'Restoring trash item(s)',
+                notificationName,
+                'started',
+                0
+            );
+
+            Y.each(e.trashItems, function (trashItem) {
+                trashItem.restore(loadOptions, tasks.add(function (error) {
+                    var trashItemId = trashItem.get('id');
+
+                    if (error) {
+                        service._notify(
+                            "An error occurred while restoring trash item with Id: " + trashItemId
+                            + ". For the moment item's whose original location has been deleted can not be restored",
+                            notificationName,
+                            'error',
+                            0
+                        );
+                    }
+                }));
+            });
+
+            tasks.done(function () {
+
+                service._notify(
+                    'Trash item(s) have been restored',
```

Comment:
I would include the number of restored items (Should be called "Content items" as well) and this notification should be thrown only if at least one item was successfully restored.